mantle/platform/qemu.go

```diff
@@ -831,6 +833,18 @@ func (builder *QemuBuilder) AddPrimaryDisk(disk *Disk) error {
 	return nil
 }
 
+// AddBootDisk sets the instance to boot only from the target disk
+func (builder *QemuBuilder) AddBootDisk(disk *Disk) {
+	if builder.primaryDisk != nil {
+		panic("Multiple primary disks specified")
```

Comment:
Please make this an error return to match the other similar functions.  Panics are hard to clean up once they've insinuated themselves into a codebase; let's not make that problem worse.

Reply:
It's a "programmer error" though, not something the user should ever see.  We do have other similar panics in the qemu code.

Reply:
> It's a "programmer error" though, not something the user should ever see.

Well, sorta.  If we ever call this based on user input (e.g. cmdline args), that forces the caller to do its own duplicate checking rather than just relying on this code to do it.

> We do have other similar panics in the qemu code.

Right, thus "let's not make that problem worse".

Reply:
> Well, sorta. If we ever call this based on user input (e.g. cmdline args), that forces the caller to do its own duplicate checking rather than just relying on this code to do it.

The commandline processing code only takes one value for `--qemu-image`, so no extra duplicate checking is required.

Reply:
In the future, I mean.

Reply:
Which is to say, I've seen too many cases where programmer errors later turn into legitimate runtime errors, and then large parts of the call chain need to be modified.  Let's just do it right from the beginning.

Reply:
I reworked this as requested.